app/src/org/commcare/utils/GeoUtils.java

```diff
@@ -131,6 +133,32 @@ private static StandardAlertDialog setupAlertFactory(Context context,
         return factory;
     }
 
+    public static void goToProperLocationSettingsScreen(Context context) {
+        Intent intent;
+        LocationManager lm = (LocationManager)context.getSystemService(Context.LOCATION_SERVICE);
+        if (locationServicesEnabledGlobally(lm)) {
+            intent = new Intent(android.provider.Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
+            intent.setData(Uri.fromParts("package", context.getPackageName(), null));
+        } else {
+            intent = new Intent(android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+        }
+        context.startActivity(intent);
+    }
+
+    private static boolean locationServicesEnabledGlobally(LocationManager lm) {
+        boolean gpsEnabled = false, networkEnabled = false;
+        try {
+            gpsEnabled = lm.isProviderEnabled(LocationManager.GPS_PROVIDER);
+        } catch (Exception ex){
```

Comment:
A comment explaining why an exception is expected/ignorable here would be useful